task-ballerina/src/task/commons.bal

```diff
@@ -54,3 +54,81 @@ public type AppointmentData record {|
+#                      fireNowWithExistingCount - Instructs the scheduler if the trigger misfires,
+#                                                            the trigger wants to be re-scheduled to 'now' with the
+#                                                            repeat count left as-is. If 'now' is after the end-time
+#                                                            the Trigger will not fire again as this does obey
+#                                                            the Trigger end-time.
+#                      fireNowWithRemainingRepeatCount - Instructs the scheduler if the trigger misfires,
+#                                                                      the SimpleTrigger wants to be re-scheduled to
+#                                                                      'now' with the repeat count set to what it
+#                                                                       would be, if it had not missed any firings.
+#                      ignorePolicy - If the trigger misfires, instructs the scheduler that the trigger will
+#                                            never be evaluated for a misfire situation, and that the scheduler will
+#                                            simply try to fire it as soon as it can, and then update the Trigger
+#                                            as if it had fired at the proper time.
+#                 For cron trigger:
+#                     smartPolicy - This is default which will act as `FireAndProceed`
+#                     ignorePolicy - If the Trigger misfires, instructs the scheduler that the trigger will
+#                                            never be evaluated for a misfire situation, and that the scheduler will
+#                                            simply try to fire it as soon as it can, and then update the trigger as
+#                                            if it had fired at the proper time.
+#                     doNothing - Instructs the scheduler If the trigger misfires, the trigger wants to have
+#                                 it's next-fire-time updated to the next time in the schedule after the current time.
+#                     fireAndProceed - Instructs the scheduler If the trigger misfires, the trigger wants to be fired
+#                                      now by Scheduler.
+public type MisfireConfiguration record {|
+    int thresholdInMillis = 5000;
+    InstructionForOneOffTrigger|InstructionForRecurringTrigger|InstructionForCronTrigger|DefaultInstruction
+    instruction = "smartPolicy";
+|};
+
+# Possible types of parameters that can be passed into the `InstructionForOneOffTrigger`.
+public type InstructionForOneOffTrigger "fireNow"|"ignorePolicy";
+
+# Possible types of parameters that can be passed into the `InstructionForRecurringTrigger`.
+public type InstructionForRecurringTrigger "fireNextWithExistingCount"|"fireNextWithRemainingCount"|
```

Comment:
Is there an especial reason for not using enums here ?

https://ballerina.io/swan-lake/learn/by-example/enums.html

Reply:
According to [chat](https://ballerina-platform.slack.com/archives/C47EAELR1/p1601982303041900?thread_ts=1601980407.040700&cid=C47EAELR1), I have changed this to finite types.